services/connector/connector_flows_test.go

```diff
@@ -0,0 +1,111 @@
+package connector
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/services/connector/commands"
+	"github.com/status-im/status-go/signal"
+)
+
+func TestRequestAccountsSwitchChainAndSendTransactionFlow(t *testing.T) {
+	db, close := createDB(t)
+	defer close()
+
+	nm := commands.NetworkManagerMock{}
+	nm.SetNetworks([]*params.Network{
+		{
+			ChainID: 0x1,
+			Layer:   1,
+		},
+		{
+			ChainID: 0x5,
+			Layer:   1,
+		},
+	})
+	rpc := commands.RPCClientMock{}
+
+	service := NewService(db, &rpc, &nm)
+
+	api := NewAPI(service)
+
+	// Try to request accounts without permission
+	request := "{\"method\":\"eth_accounts\",\"params\":[],\"origin\":\"http://testDAppURL123\",\"dAppName\":\"testDAppName\",\"dAppIconUrl\":\"http://testDAppIconUrl\"}"
+	response, err := api.CallRPC(request)
+	assert.Empty(t, response)
+	assert.Error(t, err)
+	assert.Equal(t, commands.ErrDAppIsNotPermittedByUser, err)
+
+	accountAddress := types.BytesToAddress(types.FromHex("0x6d0aa2a774b74bb1d36f97700315adf962c69fcg"))
+	expectedHash := types.BytesToHash(types.FromHex("0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"))
+
+	signal.SetMobileSignalHandler(signal.MobileSignalHandler(func(s []byte) {
+		var evt commands.EventType
+		err := json.Unmarshal(s, &evt)
+		assert.NoError(t, err)
+
+		switch evt.Type {
+		case signal.EventConnectorSendRequestAccounts:
+			var ev signal.ConnectorSendRequestAccounts
+			err := json.Unmarshal(evt.Event, &ev)
+			assert.NoError(t, err)
+
+			err = api.RequestAccountsFinished(commands.RequestAccountsFinishedArgs{
+				Accounts: []types.Address{accountAddress},
+				Error:    nil,
+			})
+			assert.NoError(t, err)
+		case signal.EventConnectorSendTransaction:
+			var ev signal.ConnectorSendTransactionSignal
+			err := json.Unmarshal(evt.Event, &ev)
+			assert.NoError(t, err)
+
+			err = api.SendTransactionFinished(commands.SendTransactionFinishedArgs{
+				Hash:  expectedHash,
+				Error: nil,
+			})
+			assert.NoError(t, err)
+		}
+	}))
+
+	// Request accounts, now for real
```

Comment:
I can imagine your pain going through the escaping, did you see if having a function for escaping the params would make sense? That way users (browser plugin) can just send the json and we do the escaping. It's just a suggestion...

Reply:
for these test datas i used chatgpt to create valid escaped jsons, I find it a little easier than calling a function from status-go. I'd also like to keep the raw test data in the test itself, so I can copy-paste it into postman to check it out. But if @felicio needs such function it's not complicated to add one

Reply:
sure, this is not blocking for me 👍🏾

Reply:
geth require escaping from us: https://discord.com/channels/1210237582470807632/1238184550300123217/1255891170383630346

Reply:
Created a following issue here: https://github.com/status-im/status-go/issues/5528
However, if i just remove escaping here, in the test - it will work fine. But if i try to pass it as plain json into `connector_callRPC` params, geth will strip it out.